__tests__/generate-sbom.test.ts

```diff
@@ -0,0 +1,46 @@
+import {Octokit} from '@octokit/core'
+import {describe, expect} from '@jest/globals'
+import {generateSBOM} from '../src/generate-sbom'
+import fs from 'fs'
+
+const mockSBOM = fs.readFileSync('./__tests__/mock-sbom.json', 'utf-8')
+
+jest.spyOn(fs, 'writeFile').mockImplementation((f, d, callback) => {
```

Comment:
Not something for now, but if you ever work in `fs` again, it may be worth taking a look at this: https://www.npmjs.com/package/mock-fs. It's a great package for mocking file storage/retrieval. I have used it in a few projects and it's worked really well 👍

Reply:
I couldn't see any example of _writing_ files with `mock-fs`. I didn't really need to mock the `read` function, so I just when with the `spy`.